### PR TITLE
OCPBUGS-18348: Add deprecation alert for DeploymentConfig

### PR DESCRIPTION
**Pull request by avivtur**
Adding an alert for deprecation of DeploymentConfig.

More info:
Jira issue: https://issues.redhat.com/browse/OCPSTRAT-118
UX design doc: https://docs.google.com/document/d/1FXxHcsC8llE5IEFJmU3RTYDQBP9-dao718pEwAzKGQw/edit

Demo:
dc-deprecation4
dc-deprecation3
dc-deprecation2
dc-deprecation
dc-deprecation-topology

Learn more URL: 
https://access.redhat.com/documentation/en-us/openshift_container_platform/4.13/html/building_applications/deployments#doc-wrapper
TODO: 

- [x] add URL for the "Learn more" link button

### CODE REVIEW

**Comment by jhadvig** [comment]
@opayne1 could you please help to formulate the info alerts text here.
TBH Im not a big fan a alert that starts with `We're` or `We'll`. Also we should be avoiding abbreviations and also this type of generalisation.

File: frontend/public/components/deployment-config.tsx
Comment (opayne1) [inline frontend/public/components/deployment-config.tsx:273]: What are your thoughts on something like:

```suggestion
          'public~Support for DeploymentConfigs will be removed from a future OpenShift Container Platform release, currently planned as OpenShift Container Platform 4.17. You should migrate to Deployments as soon as possible.',
```
In the docs we try and avoid a date or a specific future version if possible. We have gotten away with using "planned" before with supports approval.

**Review by opayne1 — COMMENTED** [review]
@jhadvig Added a suggestion! Let me know if you would like to discuss more 😄

File: frontend/public/components/deployment-config.tsx
Comment (jhadvig) [reply frontend/public/components/deployment-config.tsx:273]: Yeah, here I would also rather avoid the dates, since we are not getting rid of the DC just deprecating and dropping support. That said I dont think we will be able to remove it from the UI even after 4.17.
That said I like the wording 👍 

cc'ing @ingvagabund for additional pair of eyes 👀

Comment (ingvagabund) [reply frontend/public/components/deployment-config.tsx:273]: DeploymentConfigs are going to be supported in every 4.X version. 4.14 is the first OCP version where we officially announce the deprecation (without removal) and recommend everyone to migrate to deployments wherever possible.

Comment (ingvagabund) [reply frontend/public/components/deployment-config.tsx:273]: Something along these lines:
```
DeploymentConfigs are getting deprecated in OpenShift Container Platform 4.14. We will continue to support DeploymentConfigs, but we encourage everyone to migrate to Deployments wherever it is possible.
```

Comment (jhadvig) [reply frontend/public/components/deployment-config.tsx:273]: I like it. lets tweak it a bit with:
```
DeploymentConfigs are getting deprecated in OpenShift Container Platform 4.14. DeploymentConfigs will continue to be supported, but should be migrate to Deployments due to support.
```

Comment (jhadvig) [reply frontend/public/components/deployment-config.tsx:273]: @opayne1 thoughts?

Comment (opayne1) [reply frontend/public/components/deployment-config.tsx:273]: What do you think of this? I was going off the suggestion I made in the [google doc](https://docs.google.com/document/d/1FXxHcsC8llE5IEFJmU3RTYDQBP9-dao718pEwAzKGQw/edit#heading=h.uyky4i3aaoaf) 

> Feature development of DeploymentConfigs will be deprecated in OpenShift Container Platform 4.14. DeploymentConfigs will continue to be supported for security and critical fixes, but you should migrate to Deployments wherever it is possible.

or 

> DeploymentConfigs will be deprecated in OpenShift Container Platform 4.14. DeploymentConfigs will continue to be supported, but you should migrate to Deployments due to support.

Comment (jhadvig) [reply frontend/public/components/deployment-config.tsx:273]: @ingvagabund are we going to do security and critical fixes ?

Comment (jhadvig) [reply frontend/public/components/deployment-config.tsx:273]: Then I would go with:
`Feature development of DeploymentConfigs will be deprecated in OpenShift Container Platform 4.14. DeploymentConfigs will continue to be supported for security and critical fixes, but you should migrate to Deployments wherever it is possible.`

File: frontend/public/components/deployment-config.tsx
Comment (jhadvig) [inline frontend/public/components/deployment-config.tsx:273]: Either:
```suggestion
      title={t('public~DeploymentConfigs are being deprecated with OpenShift 4.14')}
```

or 

```suggestion
      title={t('public~DeploymentConfig is being deprecated with OpenShift 4.14')}
```

@opayne1 thoughts?

File: frontend/public/components/deployment-config.tsx
Comment (jhadvig) [inline frontend/public/components/deployment-config.tsx:282]: We should be using the `getDocumentationURL()` function.
Check:
- https://github.com/openshift/console/blob/master/frontend/public/components/build.tsx#L237
- https://github.com/openshift/console/blob/master/frontend/public/components/utils/documentation.tsx#L87

This URL would refer to downstream docs link. For upstream I would use `applications/deployments/what-deployments-are.html` path

**Review by jhadvig — CHANGES_REQUESTED** [review]
Sorry for not adding these sooner.

File: frontend/public/components/deployment-config.tsx
Comment (opayne1) [reply frontend/public/components/deployment-config.tsx:273]: I would go with the second

>  title={t('public~DeploymentConfig is being deprecated with OpenShift 4.14')}

**Comment by jhadvig** [comment]
QE Approver: 
/assign @yanpzhan
PX Approver:
/assign @RickJWagner

**Comment by jhadvig** [comment]
```
Logging in as kubeadmin
      Installing operator: "Red Hat OpenShift Serverless"
      Operator Red Hat OpenShift Serverless was not yet installed.
      Performing Serverless post installation steps
  1) "before all" hook for "Create knative workload using Container image with extrenal registry on Add page: KN-05-TC05 (example #1)"
  0 passing (3m)
  1 failing
  1) Perform actions on knative service and revision
       "before all" hook for "Create knative workload using Container image with extrenal registry on Add page: KN-05-TC05 (example #1)":
     AssertionError: Timed out retrying after 40000ms: Expected to find element: `[data-test="dropdown-menu-item-link"]`, but never found it. Queried from element: <div.pf-c-menu.pf-m-scrollable.co-namespace-dropdown__menu>
Because this error occurred during a `before all` hook we are skipping all of the remaining tests.
      at Object.selectProject (https://console-openshift-console.apps.ci-op-h5gshjvp-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=support/commands/index.ts:57116:14)
      at createKnativeServing (https://console-openshift-console.apps.ci-op-h5gshjvp-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=support/commands/index.ts:57875:28)
      at performPostInstallationSteps (https://console-openshift-console.apps.ci-op-h5gshjvp-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=support/commands/index.ts:57755:61)
      at verifyAndInstallOperator (https://console-openshift-console.apps.ci-op-h5gshjvp-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=support/commands/index.ts:57796:5)
      at verifyAndInstallKnativeOperator (https://console-openshift-console.apps.ci-op-h5gshjvp-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=support/commands/index.ts:57806:42)
      at Context.eval (https://console-openshift-console.apps.ci-op-h5gshjvp-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=support/commands/index.ts:62979:49)
```

/retest

**Review by RickJWagner — COMMENTED** [review]
Just one question.  Is peaceful light blue the correct color for a deprecation?  It seems on the 'all is well' side of the spectrum, while light red or orange might better signal 'Hey, you'd better take care of this!'
Just questioning-- hopefully our UI SMEs will have already considered this.

**Comment by yanpzhan** [comment]
@XiyunZhao will test this pr.

File: frontend/public/components/deployment-config.tsx
Comment (jhadvig) [inline frontend/public/components/deployment-config.tsx:289]: @avivtur could you change this to 
```suggestion
        text={t('public~Learn more about Deployments')}
```

**Review by jhadvig — CHANGES_REQUESTED** [review]
/lgtm cancel

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by XiyunZhao** [comment]
Hi @avivtur, I created a bug https://github.com/openshift/console/pull/12968 for tracking the alert is missing in Topology -> DeploymentConfig Details Tab, could you help to check if this bug could be fixed included in this PR?

**Comment by avivtur** [comment]
> Hi @avivtur, I created a bug #12968 for tracking the alert is missing in Topology -> DeploymentConfig Details Tab, could you help to check if this bug could be fixed included in this PR?

Hi @XiyunZhao, I'd be happy to help. 
Can you please add the link for the bug please? you attached the PR's link by accident :smile:

**Comment by avivtur** [comment]
```
ERRO[2023-07-26T17:30:06Z] Some steps failed:                           
ERRO[2023-07-26T17:30:06Z] 
  * could not run steps: step e2e-gcp-console failed: "e2e-gcp-console" pre steps failed: "e2e-gcp-console" pod "e2e-gcp-console-ipi-install-times-collection" failed: could not watch pod: the pod ci-op-hrswf409/e2e-gcp-console-ipi-install-times-collection failed after 4s (failed containers: place-entrypoint): ContainerFailed one or more containers exited
Container place-entrypoint exited with code 255, reason Error
Link to step on registry info site: https://steps.ci.openshift.org/reference/ipi-install-times-collection
Link to job on registry info site: https://steps.ci.openshift.org/job?org=openshift&repo=console&branch=master&test=e2e-gcp-console 
INFO[2023-07-26T17:30:06Z] Reporting job state 'failed' with reason 'executing_graph:step_failed:utilizing_lease:executing_test:executing_multi_stage_test' 
```
/retest

**Comment by jhadvig** [comment]
@avivtur https://issues.redhat.com/browse/OCPBUGS-17085

**Comment by jhadvig** [comment]
@avivtur it looks like there is an issue with the locales, probably you will need to rebase on top of master.

File: frontend/packages/topology/src/components/side-bar/components/SideBarHeading.tsx
Comment (jhadvig) [inline frontend/packages/topology/src/components/side-bar/components/SideBarHeading.tsx:22]:
```suggestion
        <div className="dc-deprecation-sidebar-alert">
```

**Review by jhadvig — CHANGES_REQUESTED** [review]

**Comment by avivtur** [comment]
@jhadvig I saw this test failing:
```
Running:  helm-release.feature                                                            (1 of 1)
Couldn't determine Mocha version
  Logging in as kubeadmin
  Helm Release
    ✓ Create or Select the project namespace (31229ms)
    ✓ Open the Helm tab on the navigation bar when helm charts are absent: HR-05-TC01 (4036ms)
    ✓ Create Helm Release page details: HR-05-TC02 (11131ms)
      user is switching to graph view in topology page
    ✓ Install Helm Chart from +Add Page using Form View: HR-06-TC04 (38763ms)
    ✓ Context menu options of helm release: HR-01-TC01 (6030ms)
    ✓ Open the Helm tab on the navigation bar when helm charts are present: HR-05-TC05 (5220ms)
    ✓ Filter out deployed Helm Charts: HR-05-TC06 (6609ms)
    ✓ Helm release details page: HR-05-TC13 (7067ms)
    ✓ Actions menu on Helm page after helm chart upgrade: HR-08-TC01 (9194ms)
      Change Chart version popup is displayed, so clicking on the proceed button
    ✓ Perform Upgrade action on Helm Release through Context Menu: HR-08-TC04 (9075ms)
      Change Chart version popup is displayed, so clicking on the proceed button
    ✓ Perform the helm chart upgrade for already upgraded helm chart : HR-08-TC02 (15758ms)
    1) Perform Rollback action on Helm Release through Context Menu: HR-08-TC03
    2) Delete Helm Release through Context Menu: HR-01-TC03
      You are on Topology page - Graph view
    ✓ Helm release status verification: HR-01-TC04 (78459ms)
  12 passing (8m)
  2 failing
  1) Helm Release
       Perform Rollback action on Helm Release through Context Menu: HR-08-TC03:
     AssertionError: Timed out retrying after 40000ms: Expected to find element: `g.pf-topology__group__label`, but never found it.
      at Object.getGroup (https://console-openshift-console.apps.ci-op-zkg1ss1b-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=features/helm-release.feature:49940:66)
      at Object.clickOnGroup (https://console-openshift-console.apps.ci-op-zkg1ss1b-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=features/helm-release.feature:49968:30)
      at Context.eval (https://console-openshift-console.apps.ci-op-zkg1ss1b-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=features/helm-release.feature:51253:26)
      at Context.resolveAndRunStepDefinition (https://console-openshift-console.apps.ci-op-zkg1ss1b-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=features/helm-release.feature:29097:29)
      at Context.eval (https://console-openshift-console.apps.ci-op-zkg1ss1b-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=features/helm-release.feature:28406:35)
  2) Helm Release
       Delete Helm Release through Context Menu: HR-01-TC03:
     AssertionError: Timed out retrying after 40000ms: Expected to find element: `g.pf-topology__group__label`, but never found it.
      at Object.getGroup (https://console-openshift-console.apps.ci-op-zkg1ss1b-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=features/helm-release.feature:49940:66)
      at Object.rightClickOnGroup (https://console-openshift-console.apps.ci-op-zkg1ss1b-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=features/helm-release.feature:49955:30)
      at Context.eval (https://console-openshift-console.apps.ci-op-zkg1ss1b-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=features/helm-release.feature:51239:26)
      at Context.resolveAndRunStepDefinition (https://console-openshift-console.apps.ci-op-zkg1ss1b-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=features/helm-release.feature:29097:29)
      at Context.eval (https://console-openshift-console.apps.ci-op-zkg1ss1b-75d12.xxxxxxxxxxxxxxxxxxxxxx/__cypress/tests?p=features/helm-release.feature:28406:35)
[mochawesome] Report JSON saved to /go/src/github.com/openshift/console/frontend/gui_test_screenshots/cypress_report_helm.json
  (Results)
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        14                                                                               │
  │ Passing:      12                                                                               │
  │ Failing:      2                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  5                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     8 minutes, 26 seconds                                                            │
  │ Spec Ran:     helm-release.feature                                                             │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
  (Screenshots)
  -  /go/src/github.com/openshift/console/frontend/gui_test_screenshots/cypress/scree     (1280x720)
     nshots/helm-release.feature/Helm Release -- Install Helm Chart from +Add Page us               
     ing Form View HR-06-TC04 (failed).png                                                          
  -  /go/src/github.com/openshift/console/frontend/gui_test_screenshots/cypress/scree     (1280x720)
     nshots/helm-release.feature/Helm Release -- Perform Rollback action on Helm Rele               
     ase through Context Menu HR-08-TC03 (failed).png                                               
  -  /go/src/github.com/openshift/console/frontend/gui_test_screenshots/cypress/scree     (1280x720)
     nshots/helm-release.feature/Helm Release -- Perform Rollback action on Helm Rele               
     ase through Context Menu HR-08-TC03 (failed) (attempt 2).png                                   
  -  /go/src/github.com/openshift/console/frontend/gui_test_screenshots/cypress/scree     (1280x720)
     nshots/helm-release.feature/Helm Release -- Delete Helm Release through Context                
     Menu HR-01-TC03 (failed).png                                                                   
  -  /go/src/github.com/openshift/console/frontend/gui_test_screenshots/cypress/scree     (1280x720)
     nshots/helm-release.feature/Helm Release -- Delete Helm Release through Context                
     Menu HR-01-TC03 (failed) (attempt 2).png                                                       
  (Video)
  -  Started processing:  Compressing to 32 CRF                                                     
    Compression progress:  29%
    Compression progress:  61%
    Compression progress:  95%
  -  Finished processing: /go/src/github.com/openshift/console/frontend/gui_test_scre   (38 seconds)
                          enshots/cypress/videos/helm-release.feature.mp4                           
====================================================================================================
  (Run Finished)
       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✖  helm-release.feature                     08:26       14       12        2        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  1 of 1 failed (100%)                     08:26       14       12        2        -        -
    ```
    
Any thoughts on how to fix this?

**Comment by XiyunZhao** [comment]
PR has been tested in private build, the report issue is gone
/label qe-approved

**Comment by jerolimov** [comment]
Hey @avivtur @jhadvig,

I looked into why the Helm E2e job fails. Here is what I have done so far:

When you click in the latest CI job https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_console/12968/pull-ci-openshift-console-master-e2e-gcp-console/1697364072602800128 on artifacts and follow this path to the screenshots, you can see multiple screenshots that shows that the topology crashes with this change: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_console/12968/pull-ci-openshift-console-master-e2e-gcp-console/1697364072602800128/artifacts/e2e-gcp-console/test/artifacts/gui_test_screenshots/cypress/screenshots/helm-release.feature/

image

I will check out the PR to investigate why this happened and how this is related to this change. Until then, let's hold this PR please.

/hold

File: frontend/packages/topology/src/components/side-bar/components/SideBarHeading.tsx
Comment (jerolimov) [inline frontend/packages/topology/src/components/side-bar/components/SideBarHeading.tsx:21]: The crash happens here whenever the user selects a Helm Chart. The e2e test that fails imports a Helm Chart (the Nodejs one from the Helm catalog) and automatically selects the created Helm Chart.

The problem is that a Helm chart is not k8s resource. (It's mostly saved in a Secret, but there are other options as well.)

```
TypeError: Cannot read properties of undefined (reading 'kind')

Component trace:
at SideBarHeading (http://localhost:12968/static/dev-console/code-refs/topology~overview-list-page-f3b1fecca527617e2d8b.js:3145:27)
```

In this case, `resource` is undefined. But in other cases `getData()` could also return undefined!

So I recommend adding a new variable at the beginning of the component:

```tsx
  const resourceKind = element.getData()?.resource?.kind;
```

And then using this here:

```suggestion
      {resourceKind === DeploymentConfigModel.kind && (
```

**Review by jerolimov — CHANGES_REQUESTED** [review]

File: frontend/packages/topology/src/components/side-bar/components/SideBarHeading.tsx
Comment (avivtur) [reply frontend/packages/topology/src/components/side-bar/components/SideBarHeading.tsx:21]: @jerolimov Hi there :smile: 
Thank you very much for looking into this issue and for the very detailed explanation. :100: 
I'll change my code according to it

**Comment by jerolimov** [comment]
Force pushed changes looks good to me :+1: 

/retest
/unhold

**Pull request merged**
